cacheops/redis.py

```diff
@@ -1,20 +1,37 @@
 from __future__ import absolute_import
-import warnings
+
+import os.path
+import random
+import re
 import six
 import socket
 import sys
 import traceback
-from urlparse import urlparse
+import warnings
 
-from funcy import decorator, identity, memoize
-import redis
 from django.core.exceptions import ImproperlyConfigured
 from django.utils.module_loading import import_string
-import random
+from funcy import decorator, identity, memoize
+import redis
 
 from .conf import settings
 
+handle_connection_failure = identity
+
+client_class_name = getattr(settings, 'CACHEOPS_CLIENT_CLASS', None)
+client_class = import_string(client_class_name) if client_class_name else redis.StrictRedis
+
+
+def ip(hostname):
```

Comment:
Since the Redis settings are now dicts instead of URLs, we don't have to parse the hostnames out any more.